message/message.go

```diff
@@ -499,3 +504,51 @@ func getBytes(msg interface{}) []byte {
 	}
 	return buf.Bytes()
 }
+
+func getFilePath() string {
+	filename := "unique.id"
+	folder := "/tmp/pubnub/"
+	if _, err := os.Stat(folder); os.IsNotExist(err) {
+		err = os.Mkdir(folder, 0770)
+		if err != nil {
+			panic(err)
+		}
+	}
+	return folder + filename
+}
+
+var mu sync.Mutex
+
+func isDuplicateID(uniqueID string) bool {
```

Comment:
Can we take an internal redis credentials passed from Postmaster , this redis is used to keep track of messageIDs and dupication ? 
We can set expiry for the keys as 24 hours and clear the keys once we receive both messages